packages/p2p/src/components/my-profile/my-profile.jsx

```diff
@@ -1,70 +1,308 @@
-                    <sub className='my-profile__cell-data-dimension'>{dimension}</sub>
-                </div>
-                <div className='my-profile__cell-data-sub'>
-                    <div className='my-profile__cell-data-sub-upper'>{upper_sub_value}</div>
-                    <div className='my-profile__cell-data-sub-lower'>{lower_sub_value}</div>
-                </div>
-            </div>
-            <h2 className='my-profile__cell-title'>{title}</h2>
-        </div>
-    );
-};
+    React.useEffect(() => {
+        setIsMounted(true);
+        return () => setIsMounted(false);
+    }, []);
 
-Cell.propTypes = {
-    dimension: PropTypes.string,
-    lower_sub_value: PropTypes.string,
-    title: PropTypes.string,
-    upper_sub_value: PropTypes.string,
-    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-};
+    React.useEffect(() => {
+        if (is_mounted) {
+            advertiserAccountStatus();
+            advertiserStats();
+            advertiserInfo();
+        }
+    }, [is_mounted]);
+
+    const advertiserAccountStatus = () => {
+        return new Promise(resolve => {
+            requestWS({
+                get_account_status: 1,
+            }).then(response => {
+                if (!response.error) {
+                    const { get_account_status } = response;
+                    const { authentication } = get_account_status;
+                    setHasPoa(!(authentication.document && authentication.document.status === 'none'));
+                    setHasPoi(!(authentication.identity && authentication.identity.status === 'none'));
+                }
+                resolve();
+            });
+        });
+    };
+
+    const advertiserInfo = () => {
+        return new Promise(resolve => {
+            requestWS({
+                p2p_advertiser_info: 1,
+            }).then(response => {
+                if (!response.error) {
+                    const { p2p_advertiser_info } = response;
+                    setAdvertiserInfo(p2p_advertiser_info);
+                    setNickname(p2p_advertiser_info.name);
+                    setIsLoading(false);
+                }
+                resolve();
+            });
+        });
+    };
+
+    const advertiserStats = () => {
+        return new Promise(resolve => {
+            requestWS({
+                p2p_advertiser_stats: 1,
+                id,
+            }).then(response => {
+                const { p2p_advertiser_stats } = response;
+                setStats(p2p_advertiser_stats);
+                resolve();
+            });
+        });
+    };
+
+    const handleSubmit = values => {
+        return new Promise(resolve => {
+            requestWS({
+                p2p_advertiser_update: 1,
+                contact_info: values.contact_info,
+                payment_info: values.payment_info,
+                default_advert_description: values.default_advert_description,
+            });
+            resolve();
```

Comment:
The `Promise` is resolved immediately as `requestWS` is executed asynchronously, ensure to only `resolve()` in the `then` callback (and make sure to check for errors).